lib/source/omtsource/omt_source.go

```diff
@@ -0,0 +1,73 @@
+//go:build omt
+
+package omtsource
+
+import (
+	"github.com/fosdem/fazantix/external/libomt"
+	"github.com/fosdem/fazantix/lib/config"
+	"github.com/fosdem/fazantix/lib/encdec"
+	"github.com/fosdem/fazantix/lib/layer"
+)
+
+type OmtSource struct {
+	name   string
+	frames layer.FrameForwarder
+	recv   *libomt.OmtReceive
+}
+
+func New(name string, cfg *config.OmtSourceCfg, alloc encdec.FrameAllocator) *OmtSource {
+	f := &OmtSource{name: cfg.Name}
+	f.frames.Init(
+		name,
+		&encdec.FrameInfo{
+			FrameType: encdec.RGBAFrames,
+			PixFmt:    []uint8{},
+			FrameCfg:  cfg.FrameCfg,
+		},
+		alloc,
+	)
+	return f
+}
+
+func (f *OmtSource) Start() bool {
+	recv, err := libomt.OmtReceiveCreate(f.name, libomt.Video, libomt.PreferredVideoFormatBGRA, libomt.ReceiveFlagsNone)
+	if err != nil {
+		panic("Could not create OMT receiver")
+	}
+	f.recv = recv
+	go f.receiveLoop()
+	return true
+}
+
+func (f *OmtSource) receiveLoop() {
+	for {
+		frame := f.frames.GetFrameForWriting()
+		if frame == nil {
+			panic("framedropping for ffmpeg sources not yet implemented")
```

Comment:
I think this can be handled nicely instead: just do `f.recv.Receive()` into an empty buffer and continue the loop, to drop the frame